evennia/server/profiling/profiler.py

```diff
@@ -0,0 +1,235 @@
+"""
+Evennia memory profiler, whitenoise 2023
+
+This is currently a work in progress and is not ready for production use if you have other
+options. This is meant to take on memory profiling alternatives like `memory_profiler`, but
+with the ability to understand Evennia more, have a bit more flexibility, and not require a
+third-party library for debugging memory issues.
+
+This profiler uses tracemalloc as its backend for memory analysis, and wraps some `gc`
+functionality for the user. To use this tool will require some understanding of Python's
+internals with regards to allocation and memory management.
```

Comment:
This is a good doc string, but it's missing a usage example of how to actually _use_ this tool - that is, in which situation - from inside code, from the command line, how to generate the needed tracemalloc snapshots etc. This is because people _will_ find this code and ask about how to use it otherwise ;)